packages/react-core/src/components/Timestamp/Timestamp.tsx

```diff
@@ -0,0 +1,135 @@
+  /** Content displayed in the tooltip when using the "custom" variant. */
+  content?: React.ReactNode;
+  /** Additional props passed to the tooltip. */
+  tooltipProps?: any;
+  /** A custom suffix to apply to a "default" variant tooltip. */
+  suffix?: string;
+}
+
+export interface TimestampProps extends React.HTMLProps<HTMLSpanElement> {
+  /** Displays custom content inside the timestamp, such as a relative time. This prop will
+   * override the default content that is displayed. Custom content will not be affected by
+   * any props that format or add to displayed content.
+   */
+  children?: React.ReactNode;
+  /** Class to add to the outer span. */
+  className?: string;
+  /** Applies custom formatting to the displayed date and/or time. Passing this prop in will
+   * override the dateFormat and timeFormat props.
+   */
+  customFormat?: { [key: string]: string };
+  /** Determines the default content that is displayed inside the timestamp. If a dateTime prop is
+   * not manually passed in, this prop will also construct an ISO 8601 string for the inner time element's
+   * datetime attribute. */
+  date?: Date;
+  /** Determines the format of the displayed date in the timestamp and UTC tooltip. Examples:
+   * "Full" => Tuesday, August 9, 2022;
+   * "Long" => August 9, 2022;
+   * "Medium" => Aug 9, 2022;
+   * "Short" => 8/9/22
+   */
+  dateFormat?: 'full' | 'long' | 'medium' | 'short';
+  /** Applies a custom suffix to the displayed content, usually after the time, such as a
+   * custom time zone.
+   */
+  displaySuffix?: string;
+  /** Flag for displaying the time in a 12 hour format. If this prop is not passed in,
+   * the hour format will be based on the locale prop's value.
+   */
+  is12Hour?: boolean;
+  /** Determines which locale to use in the displayed content. Defaults to the current locale
+   * when this prop is not passed in. For more information about locale's:
+   * https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Intl#Locale_identification_and_negotiation
+   */
+  locale?: string;
+  /** Determines the format of the displayed time in the timestamp and UTC tooltip. Examples:
+   * "Full" => 11:25:00 AM Eastern Daylight Time
+   * "Long" => 11:25:00 AM EDT
+   * "Medium" => 11:25:00 AM
+   * "Short" => 11:25 AM
+   */
+  timeFormat?: 'full' | 'long' | 'medium' | 'short';
+  /** Adds a tooltip to the timestamp. */
+  tooltip?: TimestampTooltip;
+}
+
+export const Timestamp: React.FunctionComponent<TimestampProps> = ({
+  children,
+  className,
+  customFormat,
+  date: dateProp = new Date(),
+  dateFormat,
+  displaySuffix = '',
+  is12Hour,
+  locale,
+  timeFormat,
+  tooltip,
+  ...props
+}: TimestampProps) => {
+  const hasTimeFormat = timeFormat && !customFormat;
+  const formatOptions = {
+    ...(dateFormat && !customFormat && { dateStyle: dateFormat }),
+    ...(customFormat && { ...customFormat }),
+    ...(is12Hour !== undefined && { hour12: is12Hour })
```

Comment:
maybe just default is12Hour to false?

Reply:
So originally I left it without a default value since in that case, the hour format could be determined by the locale, and then if consumers want to override that they can by passing is12Hour in. I don't think I really feel strongly either way, so if it'd be better to add a default value for the prop I'm all for it (maybe a default of `true` so that the hours get set to a 12 hour format?)